controllers/updateservice_controller.go

```diff
@@ -350,6 +351,38 @@ func (r *UpdateServiceReconciler) ensureTrustedClusterCA(ctx context.Context, re
 	return err
 }
 
+func (r *UpdateServiceReconciler) ensureGraphDataSHA(ctx context.Context, reqLogger logr.Logger, instance *cv1.UpdateService,
+	resources *kubeResources) (string, error){
+
+	pod := resources.graphDataPodContainer
+	if err := controllerutil.SetControllerReference(instance, pod, r.Scheme); err != nil {
+		return "", err
+	}
+
+	// Check if this pod already exists
+	found := &corev1.Pod{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
+
+	if err != nil && apiErrors.IsNotFound(err) || found.Status.Phase == "Succeeded"{
```

Comment:
`a && b || c` requires the reader to have internalized `&&` vs `||` precedence.  Can we use explicit parens, to avoid confusion?  And actually, if we're about to `create`, I'd expect:

* `err == nil`: We have a pod.  Maybe it's happy.  Maybe not.  Who cares?  Did we get a pullspec in `status`?
* `err != nil && apiErrors.IsNotFound(err)`: No pod yet.  Probably just got this UpdateService or it just got a by-tag graph-data pullspec or something.  Create the pod.
* `err != nil && !apiErrors.IsNotFound(err)`: Kube API-server is having trouble.  Bubble up to the caller.

Reply:
fixed